Jenkinsfile

```diff
@@ -63,147 +54,79 @@ pipeline {
             }
         }
 
         stage('Extract package info') {
             steps {
-                dir('metadata-dev') {
-                    git url: "$METADATA_DEV_GIT_URL"
-
-                    script {
-                        if (PACKAGE_IS_GENERATED.toBoolean()) {
-                            sh "cp $WORKSPACE/$INPUT_FILE_NAME ./test/package_orig.json"
-                        } else {
-                            unstash "$INPUT_FILE_NAME"
-                            sh "cp $INPUT_FILE_NAME ./test/package_orig.json"
-                        }
-
-                        // example of expected value: VE_TRACKER_V1.0.0_DHIS2.33.8-en
-                        PACKAGE_VERSION = sh(
-                            returnStdout: true,
-                            script: "cat $WORKSPACE/$INPUT_FILE_NAME | awk ' BEGIN { package_found = 0; } { if(package_found == 1 && index(\$1, \"name\") != 0) { gsub(/[\",]/,\"\"); print \$2; exit(0); }  if(index(\$1, \"package\") != 0) { package_found = 1; } }'"
-                        ).trim()
-
-                        // example of expected value: 2.33.8
-                        DHIS2_BRANCH_VERSION = sh(
-                            returnStdout: true,
-                            script: "grep DHIS2Version $WORKSPACE/$INPUT_FILE_NAME | awk -F '\"' '{print \$4;}' ").trim().toLowerCase()
-
-                        echo "Package version: ${PACKAGE_VERSION}"
-                        currentBuild.description = "${PACKAGE_VERSION}"
-                        echo "DHIS2 version: ${DHIS2_BRANCH_VERSION}"
+                script {
+                    if (PACKAGE_IS_EXPORTED.toBoolean()) {
+                        sh "cp $WORKSPACE/$PACKAGE_FILE ./test/package_orig.json"
+                    } else {
+                        unstash "$PACKAGE_FILE"
+                        sh "cp $PACKAGE_FILE ./test/package_orig.json"
+                    }
 
-                        def length = sh(
-                            returnStdout: true,
-                            script: "echo -n ${DHIS2_BRANCH_VERSION} | wc -c "
-                        ).trim()
+                    DHIS2_BRANCH_VERSION = sh(returnStdout: true, script: "cat $PACKAGE_FILE | jq -r \".package .DHIS2Version\"").trim()
 
-                        if ("${length}".toInteger() > 4) {
-                            echo "DHIS2 version is a patch version."
-                            CHANNEL = ""
-                        }
-                    }
+                    currentBuild.description = sh(returnStdout: true, script: "cat $PACKAGE_FILE | jq -r \".package .name\"").trim()
                 }
             }
         }
 
-        stage('Metadata validation') {
+        stage('Validate metadata') {
             steps {
                 script {
-                    //This call is done also here just in case the previous stage is skipped
+                    // In case the 'Export package' stage was skipped, the utils repo needs to be cloned again
```

Comment:
Would it make sense to just clone in a separate stage before exporting the package? And then have a condition (when {} ) in export package stage.

Reply:
I also had this in mind initially, but decided to remove the "extra" stage that was only cloning and do it only in stages that need the cloned repo. Most of the time the "Export package" stage won't be skipped, so the order of cloning in the stages makes sense to me.

Reply:
I see. This `stage('Validate metadata')` is not conditional though and runs in every build, so seems that cloning the repo is necessary. In my mind, cases like these warrant their own stages to not introduce extra "if" logic that can be easily avoided with another non-conditional stage and make the pipeline slightly cleaner and stages isolated